modules/RTC/TraceablePeerConnection.js

```diff
@@ -1224,25 +1222,31 @@ function extractSSRCMap(desc) {
             }
         }
         for (const ssrc of mLine.ssrcs) {
-            if (ssrc.attribute !== 'msid') {
+            if (!this._usesUnifiedPlan && ssrc.attribute !== 'msid') {
                 continue; // eslint-disable-line no-continue
             }
 
-            const msid = ssrc.value;
-            let ssrcInfo = ssrcMap.get(msid);
+            // Use the mediaType as key for the source map for unified plan clients since msids are not part of
+            // the standard and the unified plan SDPs do not have a proper msid attribute for the sources.
+            // Also the ssrcs for sources do not change for Unified plan clients since RTCRtpSender#replaceTrack is
+            // used for switching the tracks so it is safe to use the mediaType as the key for the TrackSSRCInfo map.
+            const key = this._usesUnifiedPlan ? mLine.type : ssrc.value;
+            const ssrcNumber = ssrc.id;
+            let ssrcInfo = ssrcMap.get(key);
 
             if (!ssrcInfo) {
                 ssrcInfo = {
                     ssrcs: [],
                     groups: [],
-                    msid
+                    msid: key
                 };
-                ssrcMap.set(msid, ssrcInfo);
+                ssrcMap.set(key, ssrcInfo);
             }
 
-            const ssrcNumber = ssrc.id;
-
-            ssrcInfo.ssrcs.push(ssrcNumber);
+            // Avoid duplicates.
+            if (!ssrcInfo.ssrcs.find(s => s === ssrcNumber)) {
+                ssrcInfo.ssrcs.push(ssrcNumber);
+            }
 
             if (groupsMap.has(ssrcNumber)) {
```

Comment:
do I understand correctly that this executes multiple times in unified if there are more than one attributes for one SSRC? does it not result in the same group pushed multiple times to `ssrcInfo.groups` ?

Reply:
Fixed it in this [PR](https://github.com/jitsi/lib-jitsi-meet/pull/1638)